subgraph/tests/keys-utils.ts

```diff
@@ -265,6 +282,21 @@ export function createKeyExtendedEvent(
     )
   )
 
+  if (createErc20TransferEvent) {
+    // mock transaction
+    keyExtendedEvent.transaction = new ethereum.Transaction(
```

Comment:
I think we are missing the ERC20 `Transfer` event here. It should be part of the transaction so that your handler can access it as well.